src/overlays/actors/ovl_Boss_Mo/z_boss_mo.c

```diff
@@ -22,299 +128,2943 @@ const ActorInit Boss_Mo_InitVars = {
+        this->actor.params = 0;
+        Actor_SetScale(&this->actor, 0.01f);
+        Collider_InitCylinder(globalCtx, &this->coreCollider);
+        Collider_SetCylinder(globalCtx, &this->coreCollider, &this->actor, &sCylinderInit);
+        if (Flags_GetClear(globalCtx, globalCtx->roomCtx.curRoom.num)) {
+            Actor_Kill(&this->actor);
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, 0.0f, -280.0f, 0.0f, 0,
+                               0, 0, -1);
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_B_HEART, -200.0f, -280.0f, 0.0f, 0, 0, 0, 0);
+            globalCtx->unk_11D30[0] = 0xFF;
+            MO_WATER_LEVEL(globalCtx) = -500;
+            return;
+        }
+        if (gSaveContext.eventChkInf[7] & 0x10) {
+            Audio_QueueSeqCmd(0x1B);
+            this->tentMaxAngle = 5.0f;
+            this->timers[0] = 50;
+        } else {
+            this->csState = MO_INTRO_WAIT;
+            this->actionState = MO_CORE_INTRO_WAIT;
+            this->actor.world.pos.x = 1000.0f;
+            this->timers[0] = 60;
+        }
+        sMorphaTent1 = (BossMo*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_BOSS_MO,
+                                                   this->actor.world.pos.x, this->actor.world.pos.y,
+                                                   this->actor.world.pos.z, 0, 0, 0, BOSSMO_TENTACLE);
+        this->actor.draw = BossMo_DrawCore;
+        this->actor.update = BossMo_UpdateCore;
+        Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_BOSS);
+    } else {
+        Actor_SetScale(&this->actor, 0.01f);
+        BossMo_SetupTentacle(this, globalCtx);
+        this->actor.colChkInfo.mass = 0xFF;
+        MO_WATER_LEVEL(globalCtx) = -50;
+        this->waterTexAlpha = 90.0f;
+        this->actor.world.pos.y = MO_WATER_LEVEL(globalCtx);
+        this->actor.prevPos = this->targetPos = this->actor.world.pos;
+        Collider_InitJntSph(globalCtx, &this->tentCollider);
+        Collider_SetJntSph(globalCtx, &this->tentCollider, &this->actor, &sJntSphInit, this->tentElements);
+        this->tentMaxAngle = 1.0f;
+    }
+}
+
+void BossMo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BossMo* this = THIS;
+
+    if (this->actor.params >= BOSSMO_TENTACLE) {
+        Collider_DestroyJntSph(globalCtx, &this->tentCollider);
+    } else {
+        Collider_DestroyCylinder(globalCtx, &this->coreCollider);
+    }
+}
+
+void BossMo_SetupTentacle(BossMo* this, GlobalContext* globalCtx) {
+    this->actionFunc = BossMo_Tentacle;
+    this->actionState = MO_TENT_WAIT;
+    this->timers[0] = 50 + (s16)Rand_ZeroFloat(20.0f);
+}
+
+void BossMo_Tentacle(BossMo* this, GlobalContext* globalCtx) {
+    // Because of the large number of variables, I have marked the real
+    // stack variables that cannot be reordered and the pads currently
+    // not being used as temps.
```

Comment:
were these comments added when the function was non matching? now that its matched its probably not important to keep I think

Reply:
This is mostly a note for future cleanup. Basically, a bunch of the temps are probably split across the pads. It's also to clarify what all the comments next to the stack variables mean.

Reply:
Also not sure we need these comments, unless there is a cleanup planned soon, because otherwise they are just going to stay for a long time while not serving a real purpose. And I'd almost argue anyone who goes into changing the variable layout of a function should be ready to deal with stack issues to keep it matching.